net/iputils/Makefile

```diff
@@ -9,12 +9,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=iputils
-PKG_VERSION:=20240117
+PKG_VERSION:=20240905
 PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://codeload.github.com/iputils/iputils/tar.gz/$(PKG_VERSION)?
-PKG_HASH:=a5d66e2997945b2541b8f780a7f5a5ec895d53a517ae1dc4f3ab762573edea9a
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE_URL:=https://codeload.github.com/iputils/iputils/tar.xz/$(PKG_VERSION)?
```

Comment:
Is there any reason why it was changed from tar.gz to tar.xz?

cc: @ynezz would be good to have some official statement from OpenWrt core members to know, which one is preferred and why.

Reply:
> Is there any reason why it was changed from tar.gz to tar.xz?

A better question would by to ask why one would use gzip when xz is available. I'll point out that even our zlib sources are [compressed with xz](https://github.com/openwrt/openwrt/blob/5eb8b7793dfdf251f6bf75ae2b1124504844c339/package/libs/zlib/Makefile#L14).

Reply:
> A better question would by to ask why one would use gzip when xz is available.

after the recent xz-utils drama, the xz format is being avoided, examples
 
https://github.com/openwrt/openwrt/commit/bab3ae2ee7656600a185f4cef11cef94389023af
https://github.com/openwrt/openwrt/commit/706f0e395f5392da0a502f1039aa01ca14849ec5

Reply:
I don't think there's any cause for that. The xz issues have been resolved, and there's no reason to think it is more of a risk than any other dependency at this point. zstd would be a good alternative, if iputils upstream was publishing releases in that format, but it's not.  Switching back to gzip does not make sense at this point.

Reply:
> Switching back to gzip does not make sense at this point.

You mean, like its impossible to remove those changes being introduced in this pull request?

Reply:
It's perfectly possible. It just doesn't make sense. xz is a better compression algorithm.

Reply:
anyway... codeload does not generate .xz tarballs and CI is failing

Reply:
> anyway... codeload does not generate .xz tarballs and CI is failing

I'm going to switch to using the .tar.xz files directly from GitHub, as suggested by @neheb below (assuming I understood him correctly)

IMO the general question of whether xz is permissible at all is best answered by the fact that substantial core components of OpenWRT are still compressed with it and there's no indication of any movement by the project to change that. Q.E.D.